saja/task_manager.py

```diff
@@ -0,0 +1,206 @@
+import json
+import uuid
+from datetime import datetime
+from enum import Enum
+from Task import Task, UrgentTask, Status, Priority
+from utils import get_valid_input
+
+class TaskManager:
+    _instance = None
+
+    """Class to manage tasks including adding, viewing, updating, and deleting."""
+
+    def __new__(cls, *args, **kwargs):
+        """
+        Ensure only one instance of TaskManager is created.
+        """
+        if cls._instance is None:
+            cls._instance = super(TaskManager, cls).__new__(cls)
+        return cls._instance
+
+    def __init__(self, data_file='data.json'):
+        """
+        Initialize the TaskManager with a specified data file.
+        
+        Parameters:
+        - data_file: The path to the JSON file storing task data.
+        """
+        if not hasattr(self, 'initialized'):  # Prevent reinitialization
+            self.data_file = data_file
+            self.tasks = []
```

Comment:
No need to update anything here. Consider using a dictionary instead of a list, with the task ID as a key.